### PR TITLE
build: add npmignore

### PR DESCRIPTION
**Pull request by blimmer**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- ~~tests and/or benchmarks are included~~
- ~~documentation is changed or added~~
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

This PR removes test files and GitHub workflows from the NPM package.

#### Before

```
 /tmp/ajv-compiler   main  npm pack
npm notice
npm notice 📦  @fastify/ajv-compiler@3.3.1
npm notice === Tarball Contents ===
npm notice 771B  .github/.stale.yml
npm notice 273B  .github/dependabot.yml
npm notice 223B  .github/tests_checker.yml
npm notice 292B  .github/workflows/ci.yml
npm notice 1.2kB LICENSE
npm notice 7.2kB README.md
npm notice 841B  index.d.ts
npm notice 2.8kB index.js
npm notice 1.5kB package.json
npm notice 1.6kB standalone.js
npm notice 0B    test/.gitkeep
npm notice 1.6kB test/duplicated-id-compile.test.js
npm notice 6.5kB test/index.test.js
npm notice 5.7kB test/plugins.test.js
npm notice 4.1kB test/standalone.test.js
npm notice 1.2kB test/types/index.test.ts
npm notice 0B    types/.gitkeep
npm notice === Tarball Details ===
npm notice name:          @fastify/ajv-compiler
npm notice version:       3.3.1
npm notice filename:      @fastify/ajv-compiler-3.3.1.tgz
npm notice package size:  10.0 kB
npm notice unpacked size: 35.8 kB
npm notice shasum:        84e12150abf9e05439af68ec37044abdafbd2b5a
npm notice integrity:     sha512-Pt4Q7ZpuyRDdP[...]PMxm5YvvfTKSw==
npm notice total files:   17
npm notice
```

#### After

```
/tmp/ajv-compiler   fix/add-npmignore  npm pack
npm notice
npm notice 📦  @fastify/ajv-compiler@3.3.1
npm notice === Tarball Contents ===
npm notice 1.2kB LICENSE
npm notice 7.2kB README.md
npm notice 841B  index.d.ts
npm notice 2.8kB index.js
npm notice 1.5kB package.json
npm notice 1.6kB standalone.js
npm notice 0B    types/.gitkeep
npm notice === Tarball Details ===
npm notice name:          @fastify/ajv-compiler
npm notice version:       3.3.1
npm notice filename:      @fastify/ajv-compiler-3.3.1.tgz
npm notice package size:  5.4 kB
npm notice unpacked size: 15.0 kB
npm notice shasum:        5da18f5f7b6ce70f11a3f080ee49c6947f59e40b
npm notice integrity:     sha512-XAXrUoW5b1Ihf[...]ceVUpiul6hWVg==
npm notice total files:   7
npm notice
```

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/.github
+/test
```
Comment (Eomm) [inline .npmignore:2]: We should copy-paste the `.gitignore` file here or the local `.vscode/` directory would be released.

Could you restore the `test/` folder?
It is an org policy to ship it within the npm package.

**Comment by Uzlopak** [comment]
We had this discussion multiple times. And we concluded, that we dont use .npmignore nor files option in package.json. We basically ship everything

@mcollina 
@jsumners

**Comment by jsumners** [comment]
@Uzlopak do you recall where the last conversation took place between myself and you? I'd rather link to my summary than repeat it.

**Comment by Uzlopak** [comment]
@jsumners 
https://github.com/fastify/skeleton/issues/42

**Comment by jsumners** [comment]
> @jsumners [fastify/skeleton#42](https://github.com/fastify/skeleton/issues/42)

🤣 that's why I couldn't find it. I thought it was you and I that had the dicussion, but it was actually @Fdawgs.

**Comment by jsumners** [comment]
Thank you for your interest in this topic. Please see the discussion in https://github.com/fastify/skeleton/issues/42#issuecomment-1208090783 for my reasoning against such a change.

-----

Above is my new canned response for these.

**Pull request closed without merging**